packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -0,0 +1,665 @@
+    _listenToController(_autocompleteController);
+
+    SchedulerBinding.instance!.addPostFrameCallback((Duration _) {
+      _updateOverlay();
+    });
+  }
+
+  @override
+  void didUpdateWidget(AutocompleteCore<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    // Handle changes in the AutocompleteController.
+    if (widget.autocompleteController == null && oldWidget.autocompleteController != null) {
+      _unlistenToController(oldWidget.autocompleteController!);
+      _autocompleteController = AutocompleteController<T>(
+        options: widget.options!,
+      );
+      _listenToController(_autocompleteController);
+    } else if (widget.autocompleteController != null && oldWidget.autocompleteController == null) {
+      _unlistenToController(_autocompleteController);
+      _autocompleteController.dispose();
+      _autocompleteController = widget.autocompleteController!;
+      _listenToController(_autocompleteController);
+    } else if (widget.autocompleteController != oldWidget.autocompleteController
+        && widget.autocompleteController != null
+        && oldWidget.autocompleteController != null) {
+      _unlistenToController(oldWidget.autocompleteController!);
+      _autocompleteController = widget.autocompleteController!;
+      _listenToController(_autocompleteController);
+    }
+
+    SchedulerBinding.instance!.addPostFrameCallback((Duration _) {
+      _updateOverlay();
+    });
+  }
+
+  @override
+  void dispose() {
+    _unlistenToController(_autocompleteController);
+    if (widget.autocompleteController == null) {
+      _autocompleteController.dispose();
+    }
+    _floatingResults?.remove();
+    _floatingResults = null;
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      key: _fieldKey,
+      child: CompositedTransformTarget(
+        link: _resultsLayerLink,
+        child: widget.buildField(
+          context,
+          _autocompleteController.textEditingController,
+          _onFieldSubmitted,
+        ),
+      ),
+    );
+  }
+}
+
+class _FloatingResults<T> extends StatelessWidget {
+  const _FloatingResults({
+    Key? key,
+    required this.buildResults,
+    required this.fieldSize,
+    required this.layerLink,
+    required this.onSelected,
+    required this.results,
+  }) : assert(buildResults != null),
+       assert(fieldSize != null),
+       assert(layerLink != null),
+       assert(onSelected != null),
+       assert(results != null),
+       super(key: key);
+
+  final AutocompleteResultsBuilder<T> buildResults;
+  final Size fieldSize;
+  final LayerLink layerLink;
+  final AutocompleteOnSelected<T> onSelected;
+  final List<T> results;
+
+  @override
+  Widget build(BuildContext context) {
+    return Positioned(
+      width: fieldSize.width,
+      child: CompositedTransformFollower(
+        link: layerLink,
+        showWhenUnlinked: false,
+        offset: Offset(
```

Comment:
We may also want to consider exposing this offset to our users so that they are able to decide where they want their dropdown to appear as well. For example, some users might want a gap/some padding between the field and the menu, others might want the dropdown to appear on the right/left side of the field.

Reply:
I agree. In fact I think that developers will want a way have more control over both where menu appears and how wide it is relative to the Autocomplete widget itself.  Maybe a callback is warranted?

Reply:
The way that I handled this in the design doc ([here](https://docs.google.com/document/d/1fV4FDNdcza1ITU7hlgweCDUZdWyCqd-rjz_J7K2KkfY/edit#heading=h.5fgb2vtjkmkq)) was with an `overlayEntryBuilder` callback.  I figured we could add it if users asked for it.

If you guys think it's important for V1 then I'll add it in this PR.  It should be pretty simple.

Reply:
I think it's OK, to add such a callback in a future PR.

Reply:
Sounds good.  I'll keep an eye out for user demand for it once this is released.